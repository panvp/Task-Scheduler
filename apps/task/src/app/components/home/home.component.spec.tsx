import { fireEvent, render, screen } from '@testing-library/react';
import Home from './home.component';

describe('Home Screen Test Cases', () => {
  beforeEach(() => {
    HTMLDialogElement.prototype.showModal = jest.fn();
    HTMLDialogElement.prototype.close = jest.fn();
    render(<Home />);
  });
  test('Snapshot testing', () => {
    const homeScreen = screen.getByTestId('homeScreen');
    expect(homeScreen).toMatchSnapshot();
  });

  test("Add Task button",()=>{
    const addButton = screen.getByTestId("addTaskButton");
    fireEvent.click(addButton);
    expect(addButton).toBeDefined();
  });
  test("Edit and delete button button",()=>{
    const addButton = screen.getByTestId("addTaskButton");
    fireEvent.click(addButton);

    const titleInput = screen.getByLabelText("Task Name:")
    fireEvent.change(titleInput, {target: {value: 'First task'}})

    const saveTaskButton = screen.getByText("Add task");
    fireEvent.click(saveTaskButton);

    const editButton = screen.getByTestId("editButton");
    fireEvent.click(editButton);

    const saveOnEditButton = screen.getByText("Update task");
    fireEvent.click(saveOnEditButton);

    const deleteButton = screen.getByTestId("deleteButton");
    fireEvent.click(deleteButton);

  });
});
