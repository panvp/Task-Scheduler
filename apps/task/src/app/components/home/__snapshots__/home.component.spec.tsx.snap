// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Test Cases Snapshot testing 1`] = `
<div
  class="bg-sky-100 h-screen flex flex-col items-center"
  data-testid="homeScreen"
>
  <div
    class="p-8 text-4xl font-bold"
  >
    Task Scheduler
  </div>
  <dialog
    class="modal-newTask"
    data-testid="newTaskComponent"
  >
    <div
      class="m-4 p-4 md:mt-4 md:mb-8 md:p-8 flex flex-col justify-around items-start"
    >
      <h1
        class="text-2xl font-bold self-center"
      >
         
        Add your task
      </h1>
      <div
        class="flex flex-wrap mt-2 md:mt-4"
      >
        <div>
          <label
            for="titleInput"
          >
             Task Name: 
          </label>
          <input
            class="sm:mx-2"
            id="titleInput"
            placeholder="Enter name here"
            size="12"
            type="text"
            value=""
          />
        </div>
        <select>
          <option>
            To Do
          </option>
          <option>
            In Progress
          </option>
          <option>
            Done
          </option>
        </select>
      </div>
      <div
        class="mt-2 md:mt-4"
      >
        <label
          class=""
          for="titleInput"
        >
           Task Description: 
        </label>
        <input
          class="sm:mx-2"
          id="titleInput"
          placeholder="Enter here"
          type="text"
          value=""
        />
      </div>
      <button
        class="font-medium rounded-lg text-sm px-2 md:px-8 py-2 mt-2 md:mt-4 text-center self-center bg-slate-200"
        disabled=""
      >
         
        Add task
      </button>
    </div>
  </dialog>
  <div
    class="taskList"
  >
    <div
      class="flex justify-between"
    >
      <button
        class="text-white bg-blue-700 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2"
        data-testid="addTaskButton"
        type="button"
      >
        Add Task
      </button>
      <select
        class="text-blue-600 font-medium rounded-lg text-sm px-5 md:px-2 py-2.5 mb-2 cursor-pointer"
        data-testid="filterButton"
      >
        <option>
           All
        </option>
        <option>
           To Do
        </option>
        <option>
           In Progress
        </option>
        <option>
           Done
        </option>
      </select>
    </div>
    <div
      class="my-8 bg-slate-200"
    >
      <div
        class="flex flex-wrap mt-4 mb-4 justify-center"
        data-testid="viewAllComponent"
      />
    </div>
  </div>
</div>
`;
